cmake/modules/FindR.cmake

```
@@ -12,15 +12,70 @@
 #  R_INCLUDE_DIRS - the R include directories
 #  R_LIBRARIES - link these to use R
 #  R_ROOT_DIR - As reported by R
+#  R_EXECUTABLE - the R executable
+#  R_SCRIPT - the Rscript executable, which runs R non-interactively
+#
 # Autor: Omar Andres Zapata Mesa 31/05/2013
+# Contributor: Blake Madden 2023-12-10
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   set(CMAKE_FIND_APPBUNDLE "LAST")
 endif()
 
+# Lists subdirectories in a directory
+# https://stackoverflow.com/questions/7787823/cmake-how-to-get-the-name-of-all-subdirectories-of-a-directory
+macro(SUBDIRLIST result curdir)
+  FILE(GLOB children RELATIVE ${curdir} ${curdir}/*)
+  set(dirlist "")
+  foreach(child ${children})
+    if(IS_DIRECTORY ${curdir}/${child})
+      list(APPEND dirlist ${child})
+    endif()
+  endforeach()
+  set(${result} ${dirlist})
+endmacro()
+
 find_program(R_EXECUTABLE NAMES R R.exe)
+find_program(R_SCRIPT NAMES Rscript Rscript.exe)
+
+# If not found and we are on Windows, try to look for it in the default installation path
+if(WIN32 AND R_EXECUTABLE MATCHES "R_EXECUTABLE-NOTFOUND")
```

```suggestion
if(MSVC AND R_EXECUTABLE MATCHES "R_EXECUTABLE-NOTFOUND")
```
Should it not be `MSVC`?

---

Well, it doesn't really matter, since it is not compiler specific (i.e. it should be the same for Ninja once implemented on Windows)